### ISSUE TITLE
Announcement: Withdraw tag for XRP, XMR, XLM and other currencies

### ISSUE THREAD

**Issue by kroitor** [ccxt/ccxt#1348]
Hey, everyone! )

We are announcing a new feature: the support for withdrawal tag, as described in this post: #1313

https://github.com/ccxt/ccxt/wiki/Manual#withdraw

The new signature of the `withdraw()` method is the following:

```
exchange.withdraw (currency, amount, address, tag = undefined, params = {})
```

The tag is usually required by exchanges for withdrawing certain currencies, like Ripple (XRP), Monero (XMR) and others, please consult exchanges' docs on that. 

This feature is still a work in progress, so it might be missing here and there, we will be happy if you report back any bugs or inconsistencies you may find with it. Currently supported exchanges: HitBTC, Cryptopia, Poloniex, Binance, Bitfinex, Bittrex. Your [contributions are very welcome](https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#how-to-contribute-code)!

The new argument makes it a **backward-incompatible (breaking) change**, please don't forget to update your code accordingly, starting from ccxt version 1.10.760.

Thank you!

**Comment by wannesdemaeght** [comment]
gotcha.

2 questions:
I know 'params = {}' is optional, is the 'tag = undefined' also optional?

IS there a way to set an arry of these settings beforehand, in an attempt to keep the lines cleaner?

Like this:

```
let withdrawParams = {currency, amount, address, tag = undefined, params = {}};
exchange.withdraw (withdrawParams)
```

Haven't found a way to make it work yet...

**Comment by kroitor** [comment]
@wannesdemaeght 

>  is the 'tag = undefined' also optional?

Yes

> IS there a way to set an arry of these settings beforehand, in an attempt to keep the lines cleaner?

In ES6 (Node 8 or higher):

```JavaScript
const params = [ currency, amount, address, tag ]
exchange.withdraw (... params)
```

Does this help?

**Comment by damicoat** [comment]
I used this updated withdraw method, transferring XLM from my bittrex wallet to my binance wallet. I was assigned a TxId, and I can find the transaction on the stellar ledger, but I did not receive the coins in my binance account. Has anyone made a successful transaction to a binance wallet that accepts a tag/memo, and did you pass additional params? The binance API docs say the memo should be accessible via the addressTag key, which I also included in the withdraw params (final argument). Thanks.
